File: go/internal/scoring/testdata/structural-v1/typescript/indexed.ts
export function indexed(values: number[], queries: number[]): number {
  const index = new Map<number, number>();
  values.forEach((value, i) => index.set(value, i));
  let found = 0;
  queries.forEach(query => { found += index.get(query) ?? 0; });
  return found;
}

File: go/internal/scoring/testdata/structural-v1/go/indexed.go
package calibration

func Indexed(values []int, queries []int) int {
	index := make(map[int]int, len(values))
	for i, value := range values {
		index[value] = i
	}
	found := 0
	for _, query := range queries {
		found += index[query]
	}
	return found
}
